src/internal/analytics/__tests__/components/analytics-funnel.test.tsx

```diff
@@ -107,25 +207,70 @@ describe('AnalyticsFunnel', () => {
     expect(FunnelMetrics.funnelSuccessful).toHaveBeenCalledTimes(1);
   });
 
-  test('calls funnelCancelled when the component unmounts after cancelling', () => {
+  test('does not call funnelSuccessful when the form is submitted without errors but not unmounted', async () => {
+    // ChildComponent is a sample component that renders a button to call funnelSubmit
+    const ChildComponent = () => {
+      const { funnelSubmit } = useFunnel();
+
+      return <button onClick={funnelSubmit}>Submit</button>;
+    };
+
+    const { getByText } = render(
+      <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={3}>
+        <ChildComponent />
+      </AnalyticsFunnel>
+    );
+    expect(FunnelMetrics.funnelSuccessful).not.toHaveBeenCalled();
+
+    fireEvent.click(getByText('Submit')); // Trigger the button click event
+
+    await new Promise(r => setTimeout(r, 1000));
+
+    expect(FunnelMetrics.funnelSuccessful).not.toHaveBeenCalled();
+  });
+
+  test('calls funnelSuccessful when the form is submitted and then later unmounted', async () => {
+    // ChildComponent is a sample component that renders a button to call funnelSubmit
+    const ChildComponent = () => {
+      const { funnelSubmit } = useFunnel();
+
+      return <button onClick={funnelSubmit}>Submit</button>;
+    };
+
+    const { unmount, getByText } = render(
+      <AnalyticsFunnel funnelType="single-page" optionalStepNumbers={[]} totalFunnelSteps={3}>
+        <ChildComponent />
+      </AnalyticsFunnel>
+    );
+    expect(FunnelMetrics.funnelSuccessful).not.toHaveBeenCalled();
+
+    fireEvent.click(getByText('Submit')); // Trigger the button click event
+
+    await new Promise(r => setTimeout(r, 1000));
+
+    unmount();
+
+    expect(FunnelMetrics.funnelSuccessful).toHaveBeenCalledTimes(1);
```

Comment:
Is the expectation that this is only called after `unmount()`? Otherwise I would add the same check before as well.

Reply:
Yes, this is the expectation. I've updated the test to turn that expectation into code.